tests/from_dict_test.py

```diff
@@ -7,68 +7,137 @@
 from tests.utils import assert_equal_data
 
-
-def test_from_dict(constructor: Constructor, request: pytest.FixtureRequest) -> None:
-    if "dask" in str(constructor) or "pyspark" in str(constructor):
-        request.applymarker(pytest.mark.xfail)
-    df = nw.from_native(constructor({"a": [1, 2, 3], "b": [4, 5, 6]}))
-    native_namespace = nw.get_native_namespace(df)
-    result = nw.from_dict({"c": [1, 2], "d": [5, 6]}, native_namespace=native_namespace)
+TEST_EAGER_BACKENDS = [
+    Implementation.POLARS,
+    Implementation.PANDAS,
+    Implementation.PYARROW,
+    "polars",
+    "pandas",
+    "pyarrow",
+]
+
+
+@pytest.mark.parametrize(
+    "backend",
+    TEST_EAGER_BACKENDS,
+)
+def test_from_dict(
+    backend: Implementation | str,
+) -> None:
+    result = nw.from_dict({"c": [1, 2], "d": [5, 6]}, backend=backend)
     expected = {"c": [1, 2], "d": [5, 6]}
     assert_equal_data(result, expected)
     assert isinstance(result, nw.DataFrame)
 
 
+@pytest.mark.parametrize(
+    "backend",
+    TEST_EAGER_BACKENDS,
+)
 def test_from_dict_schema(
-    constructor: Constructor, request: pytest.FixtureRequest
+    backend: Implementation | str,
 ) -> None:
-    if "dask" in str(constructor) or "pyspark" in str(constructor):
-        request.applymarker(pytest.mark.xfail)
     schema = {"c": nw_v1.Int16(), "d": nw_v1.Float32()}
-    df = nw_v1.from_native(constructor({"a": [1, 2, 3], "b": [4, 5, 6]}))
-    native_namespace = nw_v1.get_native_namespace(df)
     result = nw_v1.from_dict(
         {"c": [1, 2], "d": [5, 6]},
-        native_namespace=native_namespace,
+        backend=backend,
         schema=schema,  # type: ignore[arg-type]
     )
     assert result.collect_schema() == schema
 
 
-def test_from_dict_without_namespace(constructor: Constructor) -> None:
-    df = nw.from_native(constructor({"a": [1, 2, 3], "b": [4, 5, 6]})).lazy().collect()
+@pytest.mark.parametrize(
+    "backend",
+    [
+        Implementation.POLARS,
+        "polars",
+    ],
+)
+def test_from_dict_without_backend(
+    constructor: Constructor, backend: Implementation | str
+) -> None:
+    df = (
+        nw.from_native(constructor({"a": [1, 2, 3], "b": [4, 5, 6]}))
+        .lazy()
+        .collect(backend=backend)
+    )
     result = nw.from_dict({"c": df["a"], "d": df["b"]})
     assert_equal_data(result, {"c": [1, 2, 3], "d": [4, 5, 6]})
 
 
-def test_from_dict_without_namespace_invalid(
+def test_from_dict_without_backend_invalid(
     constructor: Constructor,
 ) -> None:
     df = nw.from_native(constructor({"a": [1, 2, 3], "b": [4, 5, 6]})).lazy().collect()
-    with pytest.raises(TypeError, match="namespace"):
+    with pytest.raises(TypeError, match="backend"):
         nw.from_dict({"c": nw.to_native(df["a"]), "d": nw.to_native(df["b"])})
 
 
-def test_from_dict_one_native_one_narwhals(
+def test_from_dict_with_backend_invalid() -> None:
+    with pytest.raises(ValueError, match="Unsupported `backend` value"):
```

Comment:
if this test fails, we may want to skip if it if duckdb is not installed

you can do that with
```suggestion
def test_from_dict_with_backend_invalid() -> None:
    pytest.importorskip("duckdb")
    with pytest.raises(ValueError, match="Unsupported `backend` value"):
```